src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -348,7 +352,7 @@ public function setOptions(array $options)
         ));
 
         foreach ($options as $key => $value) {
-            if (isset($validOptions[$key])) {
+            if (isset($validOptions[$key]) && false === headers_sent()) {
```

Comment:
imo you should test for `(PHP_VERSION_ID < 70200 || false=== headers_sent())`, in order to keep the original (even flawed) behavior...

But that may be dirty actually.

Reply:
I think it's actually a little dirty. Then we should do that in every modified component